gpus/python-gpu-example.html.md

```diff
@@ -0,0 +1,242 @@
+* Persistent storage is provided by a Fly Volume attached to the Machine
+* Fly GPU Machines come configured to use their GPU hardware, with NVIDIA drivers installed. You can launch a vanilla Ubuntu image and run nvidia-smi with no further setup
+* This project makes use of Fly.io IPv6 private networking. It could also be configured with a Fly Proxy service so it's available via a public Anycast or private Flycast IP address
+
+## General considerations
+There's no one right way to set up a project like this. Here are some of the considerations that went into the example:
+
+### Data storage
+Machine learning tends to involve large quantities of data. We're working with a few constraints:
+* Large Docker images (many gigabytes) can be very unwieldy to push and pull, particularly over large geographical distances.
+* The root file system of a Fly Machine is ephemeral&mdash;it's reset from its Docker image on every restart. It's also limited to 50GB on GPU-enabled Machines.
+* Fly Volumes are limited to 500GB, and are attached to a physical server. The Machine must run on the same hardware as the volume it mounts.
+
+We want to shut down GPU Machines when they're not needed, either manually with `fly machine stop`, or using the Fly Proxy autostop and autostart features, so it's not desirable to download many GB of models or libraries whenever the Machine restarts. 
+
+The compromise we use here is to generate a sub-1GB Docker image and store the project's pip packages and any downloaded data on the Fly Volume. This keeps all pip dependencies together in one venv for easy coordination and flexibility. With a well-established workload, you might make a different calculation; maybe all the projects deps actually fit in a manageable Docker image, and you can dispense with the volume storage, or some packages can we installed system-wide with apt, leaving less to manage with pip.
+
+### Compute
+The GPUs available at this time are `a100-sxm4-80gb` and `a100-pcie-40gb`, and you can use one GPU per Machine. We're not currently looking at model training on a massive scale; with careful design we can certainly do some reasonable inference on a single one of these cards. Here we're looking at running models manually so we'll stick with a single Machine, but an obvious use for Fly GPU Machines is as a "stateless" service for an app whose front end and any other components run on cheaper CPU-only Machines. This allows for independent horizontal scaling of front and back ends, as well as traffic-based capacity scaling by starting and stopping Machines.
+
+At this time, Fly GPU Machines are provisioned with the `performance-8x` CPU config and 32GB RAM by default. Playing very crudely with a language model, I found it easy to out-of-memory kill my Jupyter (Python) kernel with 32GB of RAM. If you need more RAM and fastest performance, scale up with `fly scale memory`, but losing work is annoying, so it's worth enabling swap.
```

Comment:
avoid "I"? It's not used anywhere else

Reply:
Not sure. I'm explicitly offering my experience fwiw and I don't know how general it is.

Reply:
Keeping for now